Spark_practice/spark_task_1.py

```diff
@@ -0,0 +1,133 @@
+                                        t.StructField('ordering', t.StringType(), True),
+                                        t.StructField('nconst', t.StringType(), True),
+                                        t.StructField('category', t.StringType(), True),
+                                        t.StructField('job', t.StringType(), True),
+                                        t.StructField('characters', t.StringType(), True)
+                                        ])
+
+title_principals_sp = spark.read.csv('/home/kirill/PycharmProjects/spark_task_data/title_principals.tsv',
+                                     sep='\t',
+                                     header=True,
+                                     nullValue='\\N',
+                                     schema=title_principals_schema)
+
+title_ratings_schema = t.StructType([t.StructField('tconst', t.StringType(), True),
+                                     t.StructField('averageRating', t.FloatType(), True),
+                                     t.StructField('numVotes', t.IntegerType(), True)
+                                     ])
+
+title_ratings_sp = spark.read.csv('/home/kirill/PycharmProjects/spark_task_data/title_ratings.tsv',
+                                  sep='\t',
+                                  header=True,
+                                  nullValue='\\N',
+                                  schema=title_ratings_schema)
+
+path_to_save = '/home/kirill/PycharmProjects/python_education_backub052122/Spark_practice/output/'
+
+"""All times top 10 films"""
+all_time_top_films = title_basics_sp.join(title_ratings_sp, on='tconst', how='left'). \
+    select("primaryTitle", "startYear", "numVotes", "averageRating"). \
+    where((f.col("numVotes") > 100000) & (f.col("titleType") == "movie")). \
+    orderBy("averageRating", ascending=False).limit(10)
+
+all_time_top_films.write.csv(path_to_save+'all_time_top_films', header=True, mode='overwrite')
+
+"""Last 10 years top 10 films"""
+last_decade_top_films = title_basics_sp.join(title_ratings_sp, on='tconst', how='left').\
+    select("primaryTitle", "startYear", "numVotes", "averageRating").\
+    where((f.col("numVotes") > 100000) & (f.col("titleType") == "movie") & (f.col("startYear") > 2022-10)).\
+    orderBy("averageRating", ascending=False).limit(10)
+
+last_decade_top_films.write.csv(path_to_save+'last_decade_top_films', header=True, mode='overwrite')
+
+"""60's top 10 films"""
+sixtys_top_films = title_basics_sp.join(title_ratings_sp, on='tconst', how='left').\
+    select("primaryTitle", "startYear", "numVotes", "averageRating").\
+    where((f.col("numVotes") > 100000) & (f.col("titleType") == "movie") &
+          (1960 <= f.col("startYear")) &  (f.col("startYear") < 1970)).\
+    orderBy("averageRating", ascending=False).limit(10)
+
+last_decade_top_films.write.csv(path_to_save+'sixtys_top_films', header=True, mode='overwrite')
```

Comment:
Code should be structures using classes/methods or functions

Reply:
refactored